blacksquare/crossword.py

```diff
@@ -82,7 +82,7 @@ def __init__(
         self._down = np.zeros_like(self._grid, dtype=int)
         self._words = {}
         self._parse_grid()
-
+        self._usedwords = []
```

Comment:
I don't think this will work, because it won't recognize words that have been filled outside of the `fill` functionality. Basically, tracking used words either needs to live entirely within the `fill` function (by scanning the grid before beginning, and then passing an updated list through the recursion), or it needs to keep track of all possible endpoints that can complete (and uncomplete) a word (i.e. any time a cell mutates).

It might be worth breaking this functionality into a separate PR

Reply:
Sure.  I'll split it off.  
It works in the fill() funcion (which is all I care about --- I figure that if people *sant* to repear words such as:
```
xw=Crossword(3)
xw[(1,1) = BLACK
xw[ACROSS,1] = 'jar'
xw[DOWN,1] = 'jar'
xw[DOWN,2] = 'raj'
xw[ACROSS,3] = 'raj'
```
they should be allowed to do so.

Anyway, Ill split it off.  In general, the `fill()` algorithm has enough words to choose from that duplication is rare; it's only for very small crosswords that it's a problem in practice.

Reply:
Right, I think the issue is if people want to use `.fill()` on a puzzle they've partially completed manually (which I hope would be a pretty common use case) and don't want any of those words reused.